src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+  }
+}
+
+void Reactor::DecayInventory(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    mat->Decay(context()->time());
+    inventory.Push(mat);
+  }
+}
+
+void Reactor::CombineInventory(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr base = inventory.Pop();
+    int count = inventory.count();
+    for (int i = 0; i < count; i++) {
+      cyclus::Material::Ptr m = inventory.Pop();
+      base->Absorb(m);
+    }
+
+    inventory.Push(base);
+  }
+}
+
+void Reactor::ExtractHelium(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    cyclus::CompMap c = mat->comp()->atom();
+    cyclus::compmath::Normalize(&c, mat->quantity());
+
+    cyclus::CompMap He3 = {{20030000, 1}};
+
+    // A threshold of 1e-5 was set to allow tritium_reserve inventories up to
+    // 1000kg. A 1 decade lower threshold prevents tritium_reserve inventories
+    // above 33kg.
+    cyclus::Material::Ptr helium = mat->ExtractComp(
+        c[20030000], cyclus::Composition::CreateFromAtom(He3), 1e-5);
+
+    helium_storage.Push(helium);
+    inventory.Push(mat);
+  }
+}
+
+void Reactor::RecordEvent(std::string name, std::string val) {
+  context()
+      ->NewDatum("ReactorEvents")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordOperationalInfo(std::string name, std::string val) {
+  context()
+      ->NewDatum("ReactorOperationsLog")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordStatus(std::string status, double power) {
+  context()
+      ->NewDatum("ReactorStatus")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Status", status)
+      ->AddVal("Power", power)
+      ->Record();
+}
+
+void Reactor::RecordInventories(double core, double reserve, double storage,
+                                double blanket, double helium) {
+  context()
+      ->NewDatum("ReactorInventories")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("TritiumCore", core)
+      ->AddVal("TritiumReserve", reserve)
+      ->AddVal("TritiumStorage", storage)
+      ->AddVal("LithiumBlanket", blanket)
+      ->AddVal("HeliumStorage", helium)
+      ->Record();
+}
+
+void Reactor::DepleteBlanket(double bred_tritium_mass) {
+  cyclus::Material::Ptr blanket_mat = blanket.Pop();
+
+  cyclus::CompMap b = blanket_mat->comp()->mass();
+  cyclus::compmath::Normalize(&b, blanket_mat->quantity());
+
+  // Percent (as decimal) of T which comes from Li-7 instead of Li-6
+  double Li7_contribution = 0.15;
```

Comment:
On second thought, I think we should make this a modifiable variable even with the simple model. 0.15 seems like a good default that lines up with my simple models (although PW/BL/TB can have input and I can ask Hitarth what he has found)

Reply:
And this should be in the header, you don't need to re-set this variable every time `DepleteBlanket` gets called

Reply:
Made Li7_contribution a cyclus var (user definable) and as per our discussion in ines-fusion, changed the default value from 0.15 to 0.03 to reflect the group consensus that 15% was too high.